Libraries/LibWeb/HTML/OffscreenCanvas.cpp

```diff
@@ -0,0 +1,380 @@
+/*
+ * Copyright (c) 2025, Ladybird contributors
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Tuple.h>
+#include <LibWeb/Bindings/OffscreenCanvasPrototype.h>
+#include <LibWeb/HTML/Canvas/SerializeBitmap.h>
+#include <LibWeb/HTML/OffscreenCanvas.h>
+#include <LibWeb/HTML/OffscreenCanvasRenderingContext2D.h>
+#include <LibWeb/HTML/Window.h>
+#include <LibWeb/Platform/EventLoopPlugin.h>
+#include <LibWeb/WebGL/WebGL2RenderingContext.h>
+#include <LibWeb/WebGL/WebGLRenderingContext.h>
+
+namespace Web::HTML {
+
+GC_DEFINE_ALLOCATOR(OffscreenCanvas);
+
+GC::Ref<OffscreenCanvas> OffscreenCanvas::create(JS::Realm& realm, WebIDL::UnsignedLong width,
+    WebIDL::UnsignedLong height)
+{
+    return MUST(OffscreenCanvas::construct_impl(realm, width, height));
+}
+
+// https://html.spec.whatwg.org/multipage/canvas.html#dom-offscreencanvas
+WebIDL::ExceptionOr<GC::Ref<OffscreenCanvas>> OffscreenCanvas::construct_impl(
+    JS::Realm& realm,
+    WebIDL::UnsignedLong width,
+    WebIDL::UnsignedLong height)
+{
+
+    // The new OffscreenCanvas(width, height) constructor steps are:
+    auto bitmap_or_error = Gfx::Bitmap::create(Gfx::BitmapFormat::RGBA8888, Gfx::IntSize { width, height });
+
+    if (bitmap_or_error.is_error()) {
+        return WebIDL::InvalidStateError::create(realm, MUST(String::formatted("Error in allocating bitmap: {}", bitmap_or_error.error())));
+    }
+
+    auto bitmap = bitmap_or_error.release_value();
+
+    // 1. Initialize the bitmap of this to a rectangular array of transparent black pixels of the dimensions specified by width and height.
+    // TODO: I am not sure if this is already implied by the initialization, since it is all 0s? Ask that in the GitHub PR
```

Comment:
Question here

Reply:
The Bitmap's pixels are set to `rgba(0,0,0,0)` when the Bitmap is created, so I don't believe it's necessary to set them here.